jax/core.py

```diff
@@ -1676,8 +1676,7 @@ def canonicalize_shape(shape: Shape, context: str="") -> Shape:
   try:
     return tuple(unsafe_map(_canonicalize_dimension, shape))
   except TypeError:
-    pass
-  raise _invalid_shape_error(shape, context)
+    raise _invalid_shape_error(shape, context)
```

Comment:
For what it's worth, I think the reason to avoid raising the new error within the `except` statement is that it keeps the irrelevant TypeError out of the user-facing traceback.

Reply:
Yes, you are right, I have just realized this point few minutes before.
Correct simplification should be `raise _invalid_shape_error(shape, context) from None`, and it is still not equivalent to the original code if `canonicalize_shape` is called inside a `try .. except ...`, in which the original implementation will include correct context information.